File: app/src/main/java/za/co/woolworths/financial/services/android/models/dto/OtherSkus.java
package za.co.woolworths.financial.services.android.models.dto;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class OtherSkus {

	@SerializedName("sku")
	@Expose
	public String sku;
	@SerializedName("externalColourRef")
	@Expose
	public String externalColourRef;
	@SerializedName("colourImagePath")
	@Expose
	public String colourImagePath;
	@SerializedName("price")
	@Expose
	public String price;
	@SerializedName("displayName")
	@Expose
	public String displayName;
	@SerializedName("size")
	@Expose
	public String size;
	@SerializedName("colour")
	@Expose
	public String colour;
	@SerializedName("imagePath")
	@Expose
	public String imagePath;
	@SerializedName("externalImageRefV2")
	@Expose
	public String externalImageRefV2;

	@SerializedName("wasPrice")
	@Expose
	public String wasPrice;
	public int quantity = -1;
	@SerializedName("kilogramPrice")
	@Expose
	public String kilogramPrice;

	@SerializedName("styleIdOnSale")
	@Expose
	public Boolean styleIdOnSale;
}

File: app/src/main/java/za/co/woolworths/financial/services/android/models/dto/OtherSkus.kt
package za.co.woolworths.financial.services.android.models.dto

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize

@Parcelize
data class OtherSkus(
    @JvmField
    var sku: String? = null,

    var externalColourRef: String? = null,

    var colourImagePath: String? = null,

    @JvmField
    var price: String? = null,

    var displayName: String? = null,

    @JvmField
    var size: String? = null,

    @JvmField
    var colour: String? = null,

    var imagePath: String? = null,

    var externalImageRefV2: String? = null,

    var wasPrice: String? = null,

    @JvmField
    var quantity: Int = 0,

    var kilogramPrice: String? = null,

    var styleIdOnSale: Boolean? = null
) : Parcelable

fun OtherSkus.toAddToListRequest() = AddToListRequest(
    skuID = this.sku,
    giftListId = this.sku,
    catalogRefId = this.sku
)